docs/design/debug-api.md

```diff
@@ -0,0 +1,352 @@
+		"0x4e3d3e7eee350df0ee6e94a44471ee2d22cfb174db89bbf8e6c5f6aef7b360c5"
+	],
+	"id":"1"
+}
+```
+
+#### Example Response
+```JSON
+ {
+  "jsonrpc": "2.0",
+  "id": 1,
+    "result": [
+      "0x8c0041566e0bc27efe285a9e98d0b4217a46809c",
+      "0x3ee18b2214aff97000d974cf647e7c347e8fa585",
+      "0x76364611e457b1f97cd58ffc332ddc7561a193f6"
+    ]
+ }
+```
+
+### Method description
+
+`debug_getModifiedAccountsByNumber` - Returns all accounts that have changed between the two blocks specified.
+
+#### Parameters
+startNum - number - start block number
+endNum - number - end block number. Optional, defaults to startNum
+
+#### Returns 
+addresses - array of addresses
+
+#### Example Request
+```JSON
+{
+	"jsonrpc":"2.0",
+	"method":"debug_getModifiedAccountsByNumber",
+	"params":[
+		100,
+		200
+	],
+	"id":"1"
+}
+```
+
+#### Example Response
+```JSON
+ {
+  "jsonrpc": "2.0",
+  "id": 1,
+    "result": [
+      "0x8c0041566e0bc27efe285a9e98d0b4217a46809c",
+      "0x3ee18b2214aff97000d974cf647e7c347e8fa585",
+      "0x76364611e457b1f97cd58ffc332ddc7561a193f6"
+    ]
+ }
+```
+
+## Error Codes
+
+| Error Codes |        Error message        |                              Solution                               |
+| :---------: | :-------------------------: | :-----------------------------------------------------------------: |
+|    32000    |      Transaction not found.      | Occurs when user is trying to trace non-existing transaction. |
+|    32001    | Block hash not found. |          Occurs when user is trying to get modified accounts for non-existing block hash.           |
+|    32002    | Block number not found. |          Occurs when user is trying to get modified accounts for non-existing block number.           |
+
+## Limits
+1. Trying to get modified accounts for more than ex. 100 blocks. Env. variable can be `DEBUG_MODIFIED_ACCOUNTS_LIMIT`
+
+
+## Metric Capturing
+
+Capture metrics for the following:
+
+1. Log every call to all filter API method, as a total amount.
+2. Log every success or fail for each new API method, as a total amount.
+
+## Tests
+The following test cases should be covered but additional tests would be welcome.
+
+1. Overall functionality of the methods.
+2. Test different scenarios with all possible parameter combinations.
+3. Case where transaction is not found for `debug_traceTransaction`.
+4. Case where block hash is not found for `debug_getModifiedAccountsByHash`.
+5. Case where block number is not found for `debug_getModifiedAccountsByNumber`.
+
+
+## Deployment
+
+Debug API will run alongside the already available HTTP server.
+
+## Answered Questions
+
+1. What is the purpose of Debug API and what kind of goals adding it fulfil ?
+2. What new interfaces and classes would it need ?
+3. What new endpoints will be exposed ?
+4. What kind of parameters each method accepts ?
+5. What kind of response is expected to return ?
+6. What is the expected request and response ?
+7. What limits should be introduced ? 
```

Comment:
what limits are being introduced? I did not found any limits sections or on each method.
I would recommend that the limits for each method should also be configurable.

Reply:
Limits are in their own Limit section . Also yes, fully agree that all limits should be configurable.

Reply:
I see it now, maybe I missed it since is only 1 Limit currently. will leave any other comment there.